### ISSUE TITLE
Telegram Desktop reports its audio stream's client name as "telegram-desktop"

### ISSUE THREAD

**Issue by ratijas** [telegramdesktop/tdesktop#17232]
### Steps to reproduce

1. Play music in any Telegram chat / channel, or make a call

### Expected behaviour

Pipewire / PulseAudio audio stream should be registered with a client name of the app, as displayed on a task panel etc.: "Telegram Desktop", capitalized and spaced nicely.

For example, Elisa music player does this by providing Qt's app name to libVLC: https://invent.kde.org/multimedia/elisa/-/merge_requests/296/diffs

### Actual behaviour

It's just a binary's name: "telegram-desktop".

KDE/Plasma Audio Volume applet:
![Audio Volume](https://user-images.githubusercontent.com/6737986/141391755-0c665ac8-11fe-4fae-99de-f6aee18a0183.png)

System Settings Audio KCM:
![System Settings](https://user-images.githubusercontent.com/6737986/141391685-c3629890-e316-4e56-a932-586c5e1bf75c.png)

### Operating system

Arch Linux

### Version of Telegram Desktop

3.2.3-1

### Installation source

Other (unofficial) source

**Comment by ilya-fedin** [comment]
tdesktop doesn't use PA API directly, it uses cross-platform OpenAL library and it doesn't provide such API. This is not possible to implement and I think this is not even a bug (this can be a feature request, but still, I don't see a way to implement this).

**Comment by ratijas** [comment]
In that case, it's not "Closed", it's "Blocked by upstream". Please, consider reopening.

I'm not familiar with neither audio subsystem, but I'll try my best open a new issue is a more relevant place for now.

**Comment by ilya-fedin** [comment]
I don't think there's any reason to have an issue open while nothing can be done. When this would be possible, you can create a new one. But it still would need to be a feature request, not a bug report.

**Comment by ratijas** [comment]
Well, just for tracking purposes.

You are free and welcome to change this issue's category at will :)

**Comment by ilya-fedin** [comment]
Well, I don't think we really need to track it here

And I don't have rights to change issue title, nor @Aokromes has. So it's up to you :)

**Comment by ratijas** [comment]
Nah, can't modify labels either. It was assigned as a part of a template.

**Comment by ilya-fedin** [comment]
> I'm not familiar with neither audio subsystem, but I'll try my best open a new issue is a more relevant place for now.

IIRC, libpulse tries to read app name from PULSE_PROP environment variables and glib application name, but openal sets binary name as app name forcefully. I remember I've sent them a PR a while ago fixing this, they merged it, but there were no releases so far, so it's likely it just haven't reached your distribution. So, the relevant place is your distro bugtracker, to backport my commit.

**Comment by ratijas** [comment]
> there were no releases so far, so it's likely it just haven't reached your distribution. So, the relevant place is your distro bugtracker, to backport my commit.

On Arch Linux it matches the latest tag in the repository, 4th February

    openal: 1.21.1-1

https://github.com/kcat/openal-soft/tags

**Comment by ilya-fedin** [comment]
As I just said, my commit is _after_ the latest release (tag)

**Comment by ratijas** [comment]
Oh I see https://github.com/kcat/openal-soft/pull/543

It's gonna be not earlier than Christmas, heh

**Comment by ilya-fedin** [comment]
Official binary and snap cherry-pick this commit, though
